yesod-bin/AddHandler.hs

```diff
@@ -214,3 +221,10 @@ mkHandler name pattern methods = unlines
 uncapitalize :: String -> String
 uncapitalize (x:xs) = toLower x : xs
 uncapitalize "" = ""
+
+getSrcDir :: FilePath -> IO FilePath
+getSrcDir cabal = do
+    pd <- flattenPackageDescription <$> readPackageDescription normal cabal
+    let buildInfo = allBuildInfo pd
+        srcDirs = concatMap hsSourceDirs buildInfo
+    return $ head srcDirs
```

Comment:
I'm not 100% certain, but I _think_ this is worse than it seems. It's perfectly valid to omit any `hs-source-dirs`, which defaults to "use current directory." Unless I'm totally off, what we want here is `fromMaybe "." $ listToMaybe srcDirs`.